paasta_tools/cli/cmds/spark_run.py

```diff
@@ -536,8 +542,37 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
 
     if spark_app_name is None:
         spark_app_name = "paasta_spark_run"
-
     spark_app_name += f"_{get_username()}"
+    if autogen != '-1':
+        document = f"""
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    spark: exec
+spec:
+  affinity:
+    podAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: {autogen}
+        podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+            - key: spark
+              operator: In
+              values:
+              - exec
+          topologyKey: topology.kubernetes.io/hostname
+"""
+        parsed_pod_template = yaml.load(document)
+        parsed_pod_template['metadata']['labels']['spark'] += "-" + spark_app_name
+        parsed_pod_template['spec']['affinity'] \
+            ['podAffinity']['preferredDuringSchedulingIgnoredDuringExecution'][0] \
+            ['podAffinityTerm']['labelSelector']['matchExpressions'][0]['values'][0] \
+            = parsed_pod_template['metadata']['labels']['spark']
+        f = open("/nail/home/"+get_username()+"/podTemplate.yaml", "w")
```

Comment:
Generating a file with a hardcoded path in the user home path might introduce issues with concurrent user runs or with docker containers that use a different `$USER` than a human, eg. service users or `nobody`.

Reply:
Do wonder where would be the best to write to, need to be somewhere both paasta command can access and sensible within the container itself. Guess might need to do a switch case here because service user would have permission to write elsewhere but human only have permission in to `write in nail/home/get_username`

Reply:
`/nail/tmp` should always be accessible, from both services and humans and I think also always mounted as well.

Reply:
`/tmp` should also work? https://github.com/Yelp/service_configuration_lib/blob/master/service_configuration_lib/spark_config.py#L455

Reply:
good idea! it's always good to dig these things out you forget about them :P

Reply:
think /tmp had security restrictions y/docker-restrictions, so used nail/tmp